docs/articles/documentation/why-testcafe/README.md

```diff
@@ -0,0 +1,126 @@
+
+### Hooks
+
+Keep your code clean with hooks — routines that are repeated before and after each test.
+
+## Interactive when you need it
+
+### TestCafe Studio
+
+Use a GUI test recorder that makes the power of TestCafe accessible to people with zero programming experience.
+
+### Live mode
+
+Instantly see how the changes you introduce impact the test.
+
+### Debug mode
+
+Pause your tests to examine the app and troubleshoot errors.
+
+## Community-driven
+
+### Free and Open Source
+
+TestCafe is distributed for free under the MIT license.
+
+### Actively maintained
+
+The TestCafe team is quick to respond to GitHub issues and StackOverflow questions. Community suggestions shape the future of the product.
+
+### In-house support
+
+All TestCafe Studio Pro customers (subscribers and trial users alike) receive prompt, proactive support directly from the package maintainers.
+
+
+## How Does TestCafe Work?
+
+The automation engine behind TestCafe takes over browsers and the web pages they display to simulate user activity.
+
+### The Server-Client Architecture
+
+The server-client architecture of TestCafe allows it to execute system-level and in-browser code.
+
+* TestCafe uses [high-level system APIs](https://github.com/DevExpress/testcafe-browser-tools) to launch and manage browsers. This is necessary to achieve full control over test execution.
+* TestCafe tests are full-fledged Node.js scripts. They can launch services and applications, read and write system files, make use of your favourite libraries.
+* At the same time, TestCafe is capable of performing in-browser actions with client-side automation scripts. This is how TestCafe handles asynchronous events, simulates user activity, and executes custom (user-defined) JavaScript.
+
+Our experience shows that this hybrid approach gives users the best of both worlds. Early versions of TestCafe ran entirely in the browser. This caused a number of issues. The tests’ capabilities were limited, the application often interfered with page execution, and the sub-optimal logic separation caused browser crashes. Since then, we decoupled the engine, exposed the underlying Node.js API, and implemented a test runner that executes test scripts server-side.
+
+### Page Proxying: Bridging the Server-Client Gap
+
+You may notice that when you run TestCafe, the URL in the browser’s address bar does not match that of your website. This happens because TestCafe runs an under-the-hood reverse proxy.
+
+The [testcafe-hammerhead](https://github.com/DevExpress/testcafe-hammerhead) proxy intercepts browser requests and adds automation scripts to the requested resources. It then modifies the URLs contained within the resource, so that they point to the proxy. This way, neither the client-side code, nor the resources it communicates with, can tell that the page has been modified. To conceal automation scripts from the rest of the page code, TestCafe mocks the browser API.
```

Comment:
```suggestion
The [testcafe-hammerhead](https://github.com/DevExpress/testcafe-hammerhead) proxy intercepts browser requests and adds automation scripts to the requested resources. It then modifies the URLs contained within the resource, so that they point to the proxy. This means that neither the client-side code nor the linked resources can tell that the page has been modified. To conceal automation scripts from the rest of the page code, TestCafe mocks browser API.
```

Reply:
Sorry I was 99% sure I left a comment on this particular edit... Just two questions left 😄 
* The word `linked` doesn't adequately describe the relationship between the code and the resources that it communicates with... 
* Surely, the plural of API is APIs... am I missing something here?

Would this work:

```suggestion
The [testcafe-hammerhead](https://github.com/DevExpress/testcafe-hammerhead) proxy intercepts browser requests and injects automation scripts into the requested pages. When the proxy receives data, it changes all the URLs on the resource so that they point to the proxy. This means that neither the client-side code nor other resources in use can tell that the page has been modified. To conceal automation scripts from the rest of the page code, TestCafe also intercepts some of the requests to the browser API.
```